src/node_api.cc

```diff
@@ -1024,6 +1024,27 @@ napi_status napi_delete_property(napi_env env,
   return GET_RETURN_STATUS(env);
 }
 
+NAPI_EXTERN napi_status napi_has_own_property(napi_env env,
+                                              napi_value object,
+                                              napi_value key,
+                                              bool* result) {
+  NAPI_PREAMBLE(env);
+  CHECK_ARG(env, key);
+
+  v8::Isolate* isolate = env->isolate;
+  v8::Local<v8::Context> context = isolate->GetCurrentContext();
+  v8::Local<v8::Object> obj;
+
+  CHECK_TO_OBJECT(env, context, obj, object);
+  v8::Local<v8::Value> k = v8impl::V8LocalValueFromJsValue(key);
+  RETURN_STATUS_IF_FALSE(env, k->IsName(), napi_name_expected);
```

Comment:
I'm all for letting V8 handle this, but I get a compiler error passing in a `Value` :-/

```
../src/node_api.cc:1042:60: note: in instantiation of function template
      specialization 'v8::Local<v8::Name>::Local<v8::Value>' requested here
  v8::Maybe<bool> has_maybe = obj->HasOwnProperty(context, k);
```

It looks like [V8 expects a `Name`](https://v8docs.nodesource.com/node-8.0/db/d85/classv8_1_1_object.html#ab7b7245442ca6de1e1c145ea3fd653ff).